activesupport/lib/active_support/core_ext/hash/keys.rb

```diff
@@ -19,7 +19,11 @@ def transform_keys
   def transform_keys!
     return enum_for(:transform_keys!) unless block_given?
     keys.each do |key|
-      self[yield(key)] = delete(key)
+      if keys.map(&:to_s).count(key.to_s) == 1 
```

Comment:
This is going to create a huge performance overhead, and is assuming the only transformation that could create a duplicate key is `.to_s`.